src/indexable.cr

```diff
@@ -454,16 +444,6 @@ module Indexable(T)
     size == 0 ? yield : unsafe_fetch(0)
```

Comment:
Might be worth moving it the examples to enumerable. Even if the use arrays (and show one that raises `Enumerable::EmptyError`)

Then the `:inherited:` can be used. Or, maybe this `def first` can also go. Is not that the default implementation will be more inneficient.